### PR TITLE
update xfails for main, 6.0, and 5.9

### PR DESCRIPTION
**Pull request by justice-adams-apple**
### Pull Request Description

Update xfails with recent failures

### CODE REVIEW

**Review by shahmishal — APPROVED** [review]

**Comment by justice-adams-apple** [comment]
@swift-ci test

File: projects.json [hunk 1]
```diff
@@ -3385,7 +3394,29 @@
...
+        "xfail": [
+          {
+            "issue": "rdar://125516572",
+            "compatibility": ["5.7"],
+            "branch": ["main"],
+            "configuration": "release",
+            "platform": "Darwin"
+          },
+          {
+            "issue": "rdar://125516572",
```
Comment (DougGregor) [inline projects.json:3407]: This has been fixed, and the project is passing, but the radar is a dupe

Comment (DougGregor) [reply projects.json:3407]: Here's the PR that fixed it: https://github.com/apple/swift/pull/72689

File: projects.json [hunk 2]
```diff
@@ -4764,6 +4871,21 @@
             "branch": ["release/5.9"],
             "job": ["source-compat"],
             "platform": "Darwin"
+          },
+          {
+            "issue": "rdar://125516040",
```
Comment (DougGregor) [inline projects.json:4876]: This was fixed on `main` and is passing now; here's the PR: https://github.com/apple/swift/pull/72706

**Comment by justice-adams-apple** [comment]
@swift-ci test

**Pull request merged**